File: go/two-fer/two_fer.go
// Package twofer for https://exercism.org/tracks/go/exercises/two-fer
package twofer

import (
	"fmt"
)

// ShareWith returns a string with the sharing instructions.
func ShareWith(name string) string {
	if name == "" {
		return "One for you, one for me."
	}

	return fmt.Sprintf("One for %s, one for me.", name)
}

File: c/two-fer/two_fer.c
#include "two_fer.h"
#include <assert.h>
#include <stdio.h>

void two_fer(char *buffer, const char *name) {
    const char *for_name = name ? name : "you";
    int size = sprintf(buffer, "One for %s, one for me.", for_name);

    assert(size > 0);
}
